framework/infrastructure/database/mysql/BaseRepositoryMysql.ts

```diff
@@ -169,14 +198,34 @@ export default abstract class BaseRepositoryMysql<T> implements IRepositoryGener
           const [key, value] = query.split('=');
           const [, relationValue] = key.split('.');
 
-          if (!isNaN(value)) {
-            searchableFieldsIncludes[relationValue] = value;
-          } else if (typeof value === 'string') {
-            searchableFieldsIncludes[relationValue] = {
-              $like: `${value}%`,
-            };
-          } else {
-            searchableFieldsIncludes[relationValue] = value;
+          if (!value.length) {
+            throw new APIError(`${key} - ${messages.Filter.VALUE_IS_NULL}`);
+          }
+
+          var typeAttributeName = modelAttributes[relationValue]['type'].toString();
+
+          switch (true) {
```

Comment:
switch (true)? ...

Reply:
Coloquei como true, porque neste caso eu forço sempre a validação cair em uma das condições, e alem disso cada case meu é como se fosse um if. Para não encadear vários ifs elses no código adotei essa abordagem que na minha visão fica mais organizado.